content-src/lib/selectLayoutRender.js

```diff
@@ -1,79 +1,70 @@
-import {createSelector} from "reselect";
+export const selectLayoutRender = (state, rickRollCache) => {
+  const {layout, feeds, spocs} = state;
+  let spocIndex = 0;
 
-export const selectLayoutRender = createSelector(
-  // Selects layout, feeds, spocs so that we only recompute if
-  // any of these values change.
-  [
-    state => state.DiscoveryStream.layout,
-    state => state.DiscoveryStream.feeds,
-    state => state.DiscoveryStream.spocs,
-  ],
-
-  // Adds data to each component from feeds. This function only re-runs if one of the inputs change.
-  // TODO: calculate spocs
-  function layoutRender(layout, feeds, spocs) {
```

Comment:
Only chnage here is removing selector specific code. layoutRender function stay as is and renamed to selectLayoutRender with arguments as DiscoveryStream redux state and array of cached random values